### PR TITLE
Added support for gmmk pro rev2 keyboard.

### PR DESCRIPTION
**Pull request by itarze**
## Description
as the title.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/rev2/ansi/ansi.c [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        setPinInput(SPI_MOSI_PIN);
+        setPinInput(SPI_MISO_PIN);
+
+        chThdSleepMilliseconds(10);
+
+        palSetPadMode(PAL_PORT(SPI_SCK_PIN), PAL_PAD(SPI_SCK_PIN), PAL_MODE_ALTERNATE(SPI_SCK_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST | PAL_WB32_CURRENT_LEVEL3);
+        palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), PAL_MODE_ALTERNATE(SPI_MOSI_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+        palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), PAL_MODE_ALTERNATE(SPI_MISO_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+    }
+}
```
Comment (tzarc) [inline keyboards/gmmk/pro/rev2/ansi/ansi.c:263]: Is this still needed?

Comment (wb-Joy) [reply keyboards/gmmk/pro/rev2/ansi/ansi.c:263]: Please check out this PR. #17534

Comment (tzarc) [reply keyboards/gmmk/pro/rev2/ansi/ansi.c:263]: I mean, with the changes made in #17650 is this override necessary?

Comment (wb-Joy) [reply keyboards/gmmk/pro/rev2/ansi/ansi.c:263]: @tzarc  Not necessary.

Comment (wb-Joy) [reply keyboards/gmmk/pro/rev2/ansi/ansi.c:263]: Let me remove the override first, and you can agree to merge after #17534 is resolved.

File: keyboards/gmmk/pro/rev2/ansi/keymaps/via/config.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* Copyright 2021 Glorious, LLC <salman@pcgamingrace.com>
```
Comment (tzarc) [inline keyboards/gmmk/pro/rev2/ansi/keymaps/via/config.h:1]: This file is no longer needed due to #17647.

File: keyboards/gmmk/pro/rev2/iso/keymaps/via/config.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/* Copyright 2021 Glorious, LLC <salman@pcgamingrace.com>
```
Comment (tzarc) [inline keyboards/gmmk/pro/rev2/iso/keymaps/via/config.h:1]: This file isn't needed due to #17647.

**Review by tzarc — CHANGES_REQUESTED** [review]
I'd also suggest swapping the `via` keymaps to use encoder map -- VIA has picked up support for encoders using the encoder map functionality.

**Comment by itarze** [comment]
> 

Whether there are some configuration routines?

**Comment by itarze** [comment]
All have been dealt with. Please review it again. @tzarc

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Comment by bhushan-n** [comment]
How to know if i have rev1 or rev2 gmmk pro? Would be helpful if this info is added in the readme for gmmk pro.

**Comment by adophoxia** [comment]
@bhushan-n. Bit late, but if it's any worth, Glorious stated over in a [reddit post](https://www.reddit.com/r/glorious/comments/xuzlb1/comment/ir0ggel/?utm_source=share&utm_medium=web2x&context=3) sometime today regarding both revs of the boards. 
> So, "rev2" in this case probably refers to a version of GMMK PRO with a different chipset that we were preparing just in case to make sure we are ready for chip shortage if it ever comes. Fortunately, we didn't have to go through all this in the end - today's GMMK PRO still includes STM32s, as far as I know.

**Pull request merged**